fastq_filtrator/fastq_filtrator.py

```diff
@@ -0,0 +1,116 @@
+def length_filter(seq: str, length_bounds: list or int) -> bool:
+    """Answering the question if the sequence (str) passes the length filter.
+    :param seq: a string
+    :param length_bounds: the length interval for filtering, by default (0, 2**32), including bounds.
+    If one value is indicating - this is the upper bound
+    :return: boolean value indicating if the filter is passed
+    """
+    seq_length = len(seq)
+    if len(length_bounds) == 1:
+        if seq_length <= int(length_bounds):
+            return True
+        else:
+            return False
+    elif len(length_bounds) == 2:
+        if int(length_bounds[0]) <= seq_length <= int(length_bounds[1]):
+            return True
+        else:
+            return False
+
+
+def quality_filter(quality_str: str, quality_threshold: int) -> bool:
+    """ Answering the question if nucleotide sequence passes the quality filter. The filter is not passed if
+    average quality for all nucleotides is below the threshold.
+    :param quality_str: quality line from fastq file
+    :param quality_threshold: the threshold value of the average read quality (phred33 scale), 0 by default
+    :return: boolean value indicating if the filter is passed
+    """
+    mean_quality = sum([ord(el) - 33 for el in quality_str]) / len(quality_str)
+    if mean_quality >= quality_threshold:
+        return True
+    else:
+        return False
+
+
+def gc_filter(seq: str, gc_bounds: list or int) -> bool:
+    """ Answering to the Ultimate Question of Life, the Universe, and Everything...
+    Just joking:) I'm not the Deep Thought, but only a small function that answers the question if nucleotide sequence
+    passes the GC content filter.
+    :param seq: nucleotide sequence
+    :param gc_bounds: the GC interval (in percent) for filtering (by default is (0, 100), including bounds
+    :return: boolean value indicating if the filter is passed
+    """
+    gc_content = (seq.upper().count("G") + seq.upper().count("C")) * 100 / len(seq)
+    if len(gc_bounds) == 1:
+        if gc_content <= int(gc_bounds):
+            return True
+        else:
+            return False
+    elif len(gc_bounds) == 2:
+        if int(gc_bounds[0]) <= gc_content <= int(gc_bounds[1]):
+            return True
+        else:
+            return False
```

Comment:
Можно упростить. \+ не понятно, зачем превращать bounds в int
```suggestion
        return gc_content <= gc_bounds:
    elif len(gc_bounds) == 2:
        return gc_bounds[0] <= gc_content <= gc_bounds[1]:
```